staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation_test.go

```diff
@@ -161,6 +162,40 @@ func TestStructure(t *testing.T) {
 			},
 			want: field.ErrorList{},
 		},
+		{
+			desc: "duplicate kms config name",
+			in: &config.EncryptionConfiguration{
+				Resources: []config.ResourceConfiguration{
+					{
+						Resources: []string{"secrets"},
+						Providers: []config.ProviderConfiguration{
```

Comment:
Also should check when the duplication is across providers.

Reply:
I mean test both of these:

```json
{
  "kind": "EncryptionConfiguration",
  "apiVersion": "apiserver.config.k8s.io/v1",
  "resources": [
    {
      "resources": [
        "secrets"
      ],
      "providers": [
        {
          "kms": {
            "name": "panda",
            "endpoint": "unix:///@kms-provider-1.sock"
          }
        },
        {
          "kms": {
            "name": "panda",
            "endpoint": "unix:///@kms-provider-2.sock"
          }
        }
      ]
    }
  ]
}
```

```json
{
  "kind": "EncryptionConfiguration",
  "apiVersion": "apiserver.config.k8s.io/v1",
  "resources": [
    {
      "resources": [
        "secrets"
      ],
      "providers": [
        {
          "kms": {
            "name": "panda",
            "endpoint": "unix:///@kms-provider-1.sock"
          }
        }
      ]
    },
    {
      "resources": [
        "secrets"
      ],
      "providers": [
        {
          "kms": {
            "name": "panda",
            "endpoint": "unix:///@kms-provider-2.sock"
          }
        }
      ]
    }
  ]
}
```

Reply:
done!